vignettes/advanced.Rmd

```diff
@@ -221,20 +112,174 @@ See `?ci` for more information on which information can be extracted from this f
+    prepare_all: function () 
+    reset: function () 
+    run_all: function () 
+  Private:
+    name: install
+    prepare_one: function (step) 
+    run_one: function (step) 
+    steps: list
+```
+
+## Emulate a CI run locally
+
+To emulate a real CI run locally, you can call `tic()` and the stages and steps will be executed in your local R console.
+Note that doing this will of course use your local system libraries and not the CI environment.
+
+Also some steps will not be executed as they are conditioned to run on non-interactive environments only, e.g. `add_step(covr::codcov())` added by `add_package_checks()`.
+
+```{r, eval = FALSE}
+tic::tic()
+Running script: step_rcmdcheck(warnings_are_errors = warnings_are_errors, notes_are_errors = notes_are_errors, args = args, build_args = build_args)
+── R CMD build ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+✔  checking for file ‘.../DESCRIPTION’ ...
+─  preparing ‘tic’:
+✔  checking DESCRIPTION meta-information ...
+─  installing the package to build vignettes
+✔  creating vignettes (6.6s)
+   Warning: ‘inst/doc’ file
+       ‘tic-usethis-travis.R’
+     ignored as vignettes have been rebuilt.
+     Run R CMD build with --no-build-vignettes to prevent rebuilding.
+─  checking for LF line-endings in source and make files and shell scripts
+─  checking for empty or unneeded directories
+─  building ‘tic_0.2.13.9010.tar.gz’
+   
+── R CMD check ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+─  using log directory ‘/tmp/RtmpM4zdgT/file12d42bb00e56/tic.Rcheck’ (1.2s)
+─  using R version 3.5.1 (2018-07-02)
+─  using platform: x86_64-pc-linux-gnu (64-bit)
+─  using session charset: UTF-8
+─  using options ‘--no-manual --as-cran’
+✔  checking for file ‘tic/DESCRIPTION’
+─  checking extension type ... Package
+─  this is package ‘tic’ version ‘0.2.13.9010’
+─  package encoding: UTF-8
+[...]
+```
 
 # Troubleshooting: Entering the CI build
 
-If your Travis build fails and you have no clue why (error messages are too unspecific, you cannot reproduce the problem locally), you can enter into the Travis build via `ssh`.
-To do so, your repository must be [opt-in](https://docs.travis-ci.com/user/running-build-in-debug-mode/#enabling-debug-mode) for "ssh debugging".
+If your Travis build fails and you have no clue why (error messages are too unspecific, you cannot reproduce the problem locally), you can enter into the Travis build via SSH.
+To do so, your repository must be enabled for "ssh debugging".
+This means you need to write the Travis CI team an e-mail, asking whether your repo can be [activated for debugging](https://docs.travis-ci.com/user/running-build-in-debug-mode/#enabling-debug-mode).
+Usually you get a reply within one day with some additional information on how to enter the build.
```

Comment:
"enter" is overused in this paragraph.

Also, I'm missing some detail on how this works:

1. Need to pick a job we want to troubleshoot.
2. Need to restart it (with our helper).
3. Need to SSH into it.